arch/arm/src/mx8mp/mx8mp_rptun.c

```diff
@@ -0,0 +1,370 @@
+ * Name: mx8mp_rptun_is_autostart
+ ****************************************************************************/
+
+static bool mx8mp_rptun_is_autostart(struct rptun_dev_s *dev)
+{
+  return true;
+}
+
+/****************************************************************************
+ * Name: mx8mp_rptun_is_master
+ ****************************************************************************/
+
+static bool mx8mp_rptun_is_master(struct rptun_dev_s *dev)
+{
+  return false;
+}
+
+/****************************************************************************
+ * Name: mx8mp_rptun_start
+ ****************************************************************************/
+
+static int mx8mp_rptun_start(struct rptun_dev_s *dev)
+{
+  return 0;
+}
+
+/****************************************************************************
+ * Name: mx8mp_rptun_stop
+ ****************************************************************************/
+
+static int mx8mp_rptun_stop(struct rptun_dev_s *dev)
+{
+  return 0;
+}
+
+/****************************************************************************
+ * Name: mx8mp_rptun_notify
+ ****************************************************************************/
+
+static int mx8mp_rptun_notify(struct rptun_dev_s *dev, uint32_t vqid)
+{
+  mx8mp_ipc_signal(RPTUN_IPC_CHAN_MASTER_RX);
+
+  return 0;
+}
+
+/****************************************************************************
+ * Name: mx8mp_rptun_register_callback
+ ****************************************************************************/
+
+static int mx8mp_rptun_register_callback(struct rptun_dev_s *dev,
+                                       rptun_callback_t callback,
+                                       void *arg)
+{
+  struct mx8mp_rptun_dev_s *priv = container_of(dev,
+                                 struct mx8mp_rptun_dev_s, rptun);
+
+  priv->callback = callback;
+  priv->arg      = arg;
+
+  return 0;
+}
+
+/****************************************************************************
+ * Name: mx8mp_ipc_slave_callback
+ ****************************************************************************/
+
+static void mx8mp_ipc_callback(int id, void *arg)
+{
+  ipcinfo("Rptun IPC interrupt %d\n", id);
+  if (id == RPTUN_IPC_CHAN_SLAVE_RX)
+    {
+      nxsem_post(&g_mx8mp_rx_sig);
+    }
+  else
+    {
+      DEBUGASSERT(0);
+    }
+
+  __asm volatile("dsb 0xF" ::: "memory");
+}
+
+/****************************************************************************
+ * Name: mx8mp_rptun_thread
+ ****************************************************************************/
+
+static int mx8mp_rptun_thread(int argc, char *argv[])
+{
+  struct mx8mp_rptun_dev_s *dev = &g_rptun_dev;
+
+  while (1)
+    {
+      if (dev->callback != NULL)
+        {
+          dev->callback(dev->arg, RPTUN_NOTIFY_ALL);
+        }
+
+      nxsem_wait(&g_mx8mp_rx_sig);
+      up_invalidate_dcache(0x55000000, 0x55010000);
```

Comment:
why need invalidate

Reply:
These are the locations of the vrings and the vdev buffer. It is updated by the other core and the cached data is not valid anymore.